SwiftWisdom/Core/Downloader/Downloader.swift

```diff
@@ -52,6 +56,12 @@ public final class Downloader : NSObject, URLSessionDownloadDelegate {
     // MARK: NSURLSessionDownloadDelegate
     
     public func urlSession(_ session: URLSession, downloadTask: URLSessionDownloadTask, didFinishDownloadingTo location: URL) {
+        if let httpResponse = downloadTask.response as? HTTPURLResponse, !httpResponse.isSuccess {
+            updater?(.completed(.failure(DownloaderError.httpRequestFailed)))
+            updater = nil
```

Comment:
@mayasaxena Thanks for the update, but I think that this line seems repeated with the one below now, and the early return obscures this fact. Can you leave the updater = nil assignment to the end, and put success in the else-block?